src/core/featuremodel.cpp

```diff
@@ -53,6 +53,9 @@ void FeatureModel::setCurrentLayer( QgsVectorLayer* layer )
   }
 
   mLayer = layer;
+
+  connect(mLayer,&QgsVectorLayer::destroyed,this,[=]{ mRememberings.clear(); }, Qt::UniqueConnection);
```

Comment:
Qt::UniqueConnection does not work with lambdas unfortunately... It really needs to be a slot in this case.

Reply:
And just to be on the safe side for the future if we decide to load/unload layers individually (and not always in complete projects) I'd just remove the layer which has been destroyed.

Reply:
And check your pre-commit hook ;)

Reply:
Haven't knew that. I'll change it.